src/Core/Managed/Shared/Channel/InMemoryChannel.cs

```diff
@@ -133,7 +143,14 @@ public void Flush()
         /// <param name="timeout">Timeout interval to abort sending.</param>
         public void Flush(TimeSpan timeout)
         {
-            this.transmitter.Flush(timeout);
+            if (!isDisposed)
+            {
+                this.transmitter.Flush(timeout);
+            }
+            else
+            {
+                throw new ObjectDisposedException("InMemoryChannel", "The telemetry channel has been disposed.");
```

Comment:
I've discussed with Sergey and I'll take out the throw under protest.  I maintain that a disposed object is not valid for continued use and even in the pre-existing behavior any data a user tried to add after it was disposed would be lost.  Any framework that could create such a situation would have quite a bug and we'll just be hiding it.